### PR TITLE
AMPI: Expand testing of migration callbacks during privatization test

### PR DESCRIPTION
**Pull request by evan-charmworks**

### CODE REVIEW

File: tests/ampi/privatization/test-f90-tlsglobals.f90 [hunk 1]
```diff
@@ -15,15 +15,29 @@ module test_mod_tlsglobals
       end module test_mod_tlsglobals
 
 
+      ! Define these to test that calling them works
+      subroutine about_to_migrate
```
Comment (matthiasdiener) [inline tests/ampi/privatization/test-f90-tlsglobals.f90:19]: why not just do a `print *, ...` in these functions?

**Comment by matthiasdiener** [comment]
Is it possible that this patch has caused some autobuild failures (see e.g. https://charm.cs.illinois.edu/autobuild/cur/ucx-linux-x86_64.txt) @evan-charmworks

**Comment by evan-charmworks** [comment]
Yes, I can reproduce those failures locally. They are due to libgfortran.so accessing a pointer from an Isomalloc region during atexit calls.

**Comment by evan-charmworks** [comment]
```
Thread 1 "f90-tlsglobals" received signal SIGSEGV, Segmentation fault.
_gfortrani_free_format_data (fmt=0x2aaaa0000520) at ../../../src/libgfortran/io/format.c:270
270     ../../../src/libgfortran/io/format.c: No such file or directory.
(gdb) bt
#0  _gfortrani_free_format_data (fmt=0x2aaaa0000520) at ../../../src/libgfortran/io/format.c:270
#1  0x00007ffff775ab36 in _gfortrani_free_format_hash_table (u=u@entry=0x555555b2c0d0) at ../../../src/libgfortran/io/format.c:68
#2  0x00007ffff776cc70 in close_unit_1 (u=0x555555b2c0d0, locked=locked@entry=1) at ../../../src/libgfortran/io/unit.c:730
#3  0x00007ffff776cdf2 in _gfortrani_close_units () at ../../../src/libgfortran/io/unit.c:780
#4  0x00007ffff7de5b73 in _dl_fini () at dl-fini.c:138
#5  0x00007ffff63d7041 in __run_exit_handlers (status=0, listp=0x7ffff677f718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#6  0x00007ffff63d713a in __GI_exit (status=<optimized out>) at exit.c:139
#7  0x0000555555759400 in LrtsExit (exitcode=0) at /home/evan/Charmworks/charm/src/arch/mpi/machine.C:1321
#8  0x00005555557581c6 in realConverseExit (exitcode=0) at /home/evan/Charmworks/charm/src/conv-core/../arch/util/machine-common-core.C:1719
#9  0x0000555555740d6c in _exitHandler (env=0x555555cbc430) at /home/evan/Charmworks/charm/src/ck-core/init.C:672
#10 0x000055555575eb2c in CmiHandleMessage (msg=0x555555cbc430) at /home/evan/Charmworks/charm/src/conv-core/convcore.C:1708
#11 0x000055555575edce in CsdScheduleForever () at /home/evan/Charmworks/charm/src/conv-core/convcore.C:1961
#12 0x000055555575ecdc in CsdScheduler (maxmsgs=-1) at /home/evan/Charmworks/charm/src/conv-core/convcore.C:1894
#13 0x00005555557580f0 in ConverseRunPE (everReturn=0) at /home/evan/Charmworks/charm/src/conv-core/../arch/util/machine-common-core.C:1609
#14 0x0000555555757ff3 in ConverseInit (argc=1, argv=0x555555b301d0, fn=0x555555742317 <_initCharm(int, char**)>, usched=0, initret=0) at /home/evan/Charmworks/charm/src/conv-core/../arch/util/machine-common-core.C:1524
#15 0x00005555557433fd in charm_main (argc=1, argv=0x555555b301d0) at /home/evan/Charmworks/charm/src/ck-core/init.C:1819
#16 0x0000555555743428 in charm_main_fortran_wrapper_ (argc=0x7fffffffdbc8, argv=0x555555b301d0) at /home/evan/Charmworks/charm/src/ck-core/init.C:1826
#17 0x000055555562ca77 in charmxx () at /home/evan/Charmworks/charm/src/ck-core/mainf.f90:27
#18 0x000055555562cb52 in main (argc=1, argv=0x7fffffffe093) at /home/evan/Charmworks/charm/src/ck-core/mainf.f90:32
#19 0x00007ffff63b5b97 in __libc_start_main (main=0x55555562cb1c <main>, argc=1, argv=0x7fffffffdcf8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffdce8) at ../csu/libc-start.c:310
#20 0x000055555557965a in _start ()
(gdb) x/1i $rip
=> 0x7ffff775aa84 <_gfortrani_free_format_data+20>:     mov    0x50(%rdi),%eax
(gdb) p/x $rdi
$1 = 0x2aaaa0000520
```

File: tests/ampi/privatization/test-f90-tlsglobals.f90 (hunk 1 above)
Comment (matthiasdiener) [reply tests/ampi/privatization/test-f90-tlsglobals.f90:18]: I think this might have caused the underlying errors?

**Pull request merged**